File: VB/WebSite/App_Code/Grid/Binding/ObjectData_DataClass.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
	Public Class MyDataClass
		Public Function GetMyData(ByVal StartRow As Integer, ByVal PageSize As Integer) As DataView
			Return GetData(StartRow, PageSize)
		End Function
		Private Function GetData(ByVal StartRow As Integer, ByVal PageSize As Integer) As DataView
			Dim Contacts As DataSet = New DataSet()
			Contacts.ReadXml(HttpContext.Current.Server.MapPath("~/App_Data/ContactsList.xml"))

			Dim ContactsTable As DataTable = Contacts.Tables(0)
			Dim PagedContactsTable As DataTable = ContactsTable.Clone()

			Dim i As Integer = StartRow
			Do While i < StartRow + PageSize AndAlso i < ContactsTable.Rows.Count
				PagedContactsTable.ImportRow(ContactsTable.Rows(i))
				i += 1
			Loop

			Return PagedContactsTable.DefaultView
		End Function
		Public Function GetRowsCount() As Integer
			Dim Contacts As DataSet = New DataSet()
			Contacts.ReadXml(HttpContext.Current.Server.MapPath("~/App_Data/ContactsList.xml"))
			Return Contacts.Tables(0).Rows.Count
		End Function
	End Class

File: CS/WebSite/App_Code/Grid/Binding/ObjectData_DataClass.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
    public class MyDataClass {
        public DataView GetMyData(int StartRow, int PageSize) {
            return GetData(StartRow, PageSize);
        }
        private DataView GetData(int StartRow, int PageSize) {
            DataSet Contacts = new DataSet();
            Contacts.ReadXml(HttpContext.Current.Server.MapPath("~/App_Data/ContactsList.xml"));

            DataTable ContactsTable = Contacts.Tables[0];
            DataTable PagedContactsTable = ContactsTable.Clone();

            for(int i = StartRow; i < StartRow + PageSize && i < ContactsTable.Rows.Count; i++) {
                PagedContactsTable.ImportRow(ContactsTable.Rows[i]);
            }

            return PagedContactsTable.DefaultView;
        }
        public int GetRowsCount() {
            DataSet Contacts = new DataSet();
            Contacts.ReadXml(HttpContext.Current.Server.MapPath("~/App_Data/ContactsList.xml"));
            return Contacts.Tables[0].Rows.Count;
        }
    }


